tensorflow/python/ops/metrics_impl.py

```diff
@@ -259,7 +259,7 @@ def _streaming_confusion_matrix(labels, predictions, num_classes, weights=None):
     update_op: An operation that increments the confusion matrix.
   """
   # Local variable to accumulate the predictions in the confusion matrix.
-  cm_dtype = dtypes.int64 if weights is not None else dtypes.float64
+  cm_dtype = dtypes.float64 if weights is not None else dtypes.int64
```

Comment:
Can you make this float64 unconditionally? That is a slightly less breaking change, since it doesn't change the return type in the case that actually works (without weights). Changing the type in the case that was broken is a plain bugfix.

Reply:
This was the initial implementation per https://github.com/tensorflow/tensorflow/pull/12714/commits/4b521d0992fbdd7d9f18a708111e2501689f7994. @alextp explicitly asked for it to be changed to the current implementation per https://github.com/tensorflow/tensorflow/pull/12714/commits/4b521d0992fbdd7d9f18a708111e2501689f7994#r137026663.

I'd be happy to change it back to my original implementation (with the FIXME?), but please confirm that this is as desired.

In some moral sense, `int64` probably is more correct with no weights (or with integer weights), but I share your concerns about API breakage. Just want to confirm this given the specific earlier request to use `int64` in the no-weights case.

Reply:
There are no semver implications here per se, since `_streaming_confusion_matrix` is not part of the public API, and in fact no public APIs change.

Reply:
In every place we use it, we convert to float. So it may make more sense to always have it in float so we don't have to do that. 

But you're right, technically it's an implementation detail and nobody would need to ever know.

Reply:
Yeah. I'm fine either way though. Just let me know which you want.

Reply:
Let's go with all float.

Reply:
Done.